### DISCUSSION TITLE
How do i set a custom node/nvm provider?

### DISCUSSION THREAD

**Discussion by Lite5h4dow** [williamboman/nvim-lsp-installer#593] [Q&A] [answered]
i use nvm and have set nvim's provider using `fn.system("source /usr/share/nvm/init-nvm.sh &> /dev/null && echo $(dirname $(nvm which node))")` and was wondering how i can do the same for nvm-lsp-installer

**Comment by Lite5h4dow** [comment, marked as answer]
i just installed node under nvim. its solved the problem by kicking the can down the road :P

**Reply by williamboman** [reply]
Hey! At the moment, whatever `node` (and `npm`) executable that is found first in the `PATH` is used - so you'd have to activate the nvm environment prior to starting neovim. Note that this doesn't only apply to installation of servers but also the running of them! I'm not aware of any builtin mechanism of controlling this, apart from perhaps `:h provider` (which is primarily built around remote plugins as I understand it).
